fvm/src/gas/price_list.rs

```diff
@@ -636,12 +694,30 @@ pub fn price_list_by_network_version(network_version: NetworkVersion) -> &'stati
 }
 
 impl Rules for WasmGasPrices {
-    fn instruction_cost(&self, _instruction: &Instruction) -> Option<u64> {
-        Some(self.exec_instruction_cost_milli)
+    fn instruction_cost(&self, instruction: &Instruction) -> Option<u64> {
+        if self.exec_instruction_cost_milli == 0 {
+            return None;
+        }
+
+        // Rules valid for nv16. We will need to be generic over Rules (massive
+        // generics tax), use &dyn Rules (which breaks other things), or pass
+        // in the network version, or rules version, to vary these prices going
+        // forward.
```

Comment:
@Stebalien something to take into account.

Reply:
In this case, we can likely do `&'static dyn Rules`. Or, if we can't statically allocate, `Box<dyn Rules>`.